src/platform/packages/private/kbn-esql-editor/src/esql_editor.tsx

```diff
@@ -393,11 +413,17 @@ const ESQLEditorInternal = function ESQLEditor({
     );
   }, [onMouseDownResize, editorHeight, onKeyDownResize, setEditorHeight]);
 
-  const onEditorFocus = useCallback(() => {
-    setIsCodeEditorExpandedFocused(true);
-    showSuggestionsIfEmptyQuery();
-    setLabelInFocus(true);
-  }, [showSuggestionsIfEmptyQuery]);
+  const onEditorFocus = useCallback(
```

Comment:
I'm still not clear on why this method is called onKeyDown . 
Probably for some race condition to `setIsCodeEditorExpandedFocused(true)` ;,  `setLabelInFocus(true);`  ?